# PROART: Análise de Pesquisa Organizacional

**Objetivo:** Ler, Organizar e Tratar os dados.

In [18]:
# instalação de bibliotecas 
# !pip install pandas openpyxl

In [19]:
# Importação das bibliotecas necessárias para a leitura e transformação e limpeza dos dados
import pandas as pd

In [20]:
# Carregar base de dados brutos
df = pd.read_excel('data/dadosbrutos.xlsx')
df.head()

,ID de resposta,Status de resposta,Insira abaixo o código referente a seu cargo e área,Turno,Gênero,Outro,Cor/raça autodeclarada,Faixa etária,Escolaridade,Tempo na empresa,...,Insensibilidade em relação aos colegas,Dores nas pernas,Dores nas costas,Dores no corpo,Dores no braço,Distúrbios Circulatórios,Dor de cabeça,Distúrbios digestivos,Alterações no sono,Alterações no apetite
0,210878327,Started,1,NaN,NaN,NaN,NaN,NaN,NaN,NaN,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
1,210879164,Completed,6,1.0,1.0,NaN,4.0,4.0,7.0,2.0,...,1.0,2.0,2.0,2.0,2.0,1.0,2.0,2.0,1.0,1.0
2,210879349,Started,1,NaN,NaN,NaN,NaN,NaN,NaN,NaN,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
3,210879373,Started,1,1.0,1.0,NaN,1.0,4.0,7.0,5.0,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
4,210888826,Completed,6,1.0,2.0,NaN,4.0,4.0,7.0,2.0,...,3.0,4.0,3.0,2.0,2.0,3.0,2.0,2.0,3.0,3.0


In [21]:
# verificar quantidade de linhas e colunas
print(f"DIMENSÕES DOS DADOS: Linhas: {df.shape[0]} e Colunas: {df.shape[1]}")

DIMENSÕES DOS DADOS: Linhas: 473 e Colunas: 102


#### 1. Transformar as colunas numericas para categóricas

 Utilizando os dicionários de mapeamento fornecidos. Isso tornará a análise mais intuitiva e facilitará a interpretação dos resultados.


In [22]:
# Dicionários de mapeamento
map_turno = {"1": "Administrativo", "2": "Turno rotativo"}
map_genero = {"1": "Homem", "2": "Mulher", "3": "Pessoa não-binária", "4": "Prefiro não responder", "5": "Outro"}
map_cor = {"1": "Branca", "2": "Preta", "3": "Amarela", "4": "Parda", "5": "Indígena", "6": "Prefiro não responder"}
map_faixa = {"1": "Menos de 19 anos", "2": "19 a 24 anos", "3": "25 a 34 anos", "4": "35 a 44 anos", "5": "45 a 54 anos", "6": "55 a 64 anos", "7": "65 anos ou mais"}
map_escolaridade = {
    "1": "Ensino fundamental incompleto", "2": "Ensino fundamental completo", "3": "Ensino médio incompleto",
    "4": "Ensino médio completo", "5": "Ensino superior incompleto", "6": "Ensino superior completo", "7": "Pós graduação completa"
}
map_tempo_empresa = {
    "1": "Menos de 1 ano", "2": "De 1 a 3 anos", "3": "De 4 a 6 anos", "4": "De 7 a 10 anos", "5": "De 11 a 15 anos", "6": "Mais de 15 anos"
}
map_tempo_funcao = {
    "1": "Menos de 1 ano", "2": "De 1 a 3 anos", "3": "De 4 a 6 anos", "4": "De 7 a 10 anos", "5": "De 11 a 15 anos", "6": "Mais de 15 anos"
}

# Aplicar mapeamento nas colunas

# Função padrão de limpeza
def limpar_coluna(col):
    return (
        col.astype(str)
        .str.replace('.0', '', regex=False)
        .str.strip()
    )

# Aplicar mapeamento nas colunas
df['Turno'] = limpar_coluna(df['Turno']).replace(map_turno)
df['Gênero'] = limpar_coluna(df['Gênero']).replace(map_genero)
df['Cor/raça autodeclarada'] = limpar_coluna(df['Cor/raça autodeclarada']).replace(map_cor)
df['Faixa etária'] = limpar_coluna(df['Faixa etária']).replace(map_faixa)
df['Escolaridade'] = limpar_coluna(df['Escolaridade']).replace(map_escolaridade)
df['Tempo na empresa'] = limpar_coluna(df['Tempo na empresa']).replace(map_tempo_empresa)
df['Tempo na função atual dentro da empresa'] = limpar_coluna(df['Tempo na função atual dentro da empresa']).replace(map_tempo_funcao)
df.head()

,ID de resposta,Status de resposta,Insira abaixo o código referente a seu cargo e área,Turno,Gênero,Outro,Cor/raça autodeclarada,Faixa etária,Escolaridade,Tempo na empresa,...,Insensibilidade em relação aos colegas,Dores nas pernas,Dores nas costas,Dores no corpo,Dores no braço,Distúrbios Circulatórios,Dor de cabeça,Distúrbios digestivos,Alterações no sono,Alterações no apetite
0,210878327,Started,1,nan,nan,NaN,nan,nan,nan,nan,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
1,210879164,Completed,6,Administrativo,Homem,NaN,Parda,35 a 44 anos,Pós graduação completa,De 1 a 3 anos,...,1.0,2.0,2.0,2.0,2.0,1.0,2.0,2.0,1.0,1.0
2,210879349,Started,1,nan,nan,NaN,nan,nan,nan,nan,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
3,210879373,Started,1,Administrativo,Homem,NaN,Branca,35 a 44 anos,Pós graduação completa,De 11 a 15 anos,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
4,210888826,Completed,6,Administrativo,Mulher,NaN,Parda,35 a 44 anos,Pós graduação completa,De 1 a 3 anos,...,3.0,4.0,3.0,2.0,2.0,3.0,2.0,2.0,3.0,3.0


#### Observação no metadados

Vou criar 2 colunas a partir da coluna 'Insira abaixo o código referente a seu cargo e área', para facilitar a análise de área e cargo. No final remover a coluna original.

In [23]:
# CRIAR COLUNAS
# ==========================================
df['area'] = df['Insira abaixo o código referente a seu cargo e área']
df['cargo'] = df['Insira abaixo o código referente a seu cargo e área']

# DICIONÁRIOS DE MAPEAMENTO
# ==========================================
map_area = {
    "1": "Operação",
    "2": "Operação",
    "3": "Operação",
    "4": "Manutenção",
    "5": "Manutenção",
    "6": "Manutenção",
    "7": "Logística",
    "8": "Logística",
    "9": "Logística",
    "10": "RH",
    "11": "RH",
    "12": "RH",
    "13": "Segurança do Trabalho",
    "14": "Segurança do Trabalho",
    "15": "Segurança do Trabalho",
    "16": "Administrativo",
    "17": "Administrativo",
    "18": "Administrativo",
    "19": "Diretoria"
}

map_cargo = {
    "1": "Operador(a) de Produção",
    "2": "Líder de Produção",
    "3": "Supervisor(a) Operacional",
    "4": "Técnico(a) de Manutenção",
    "5": "Planejador(a) de Manutenção",
    "6": "Supervisor(a) de Manutenção",
    "7": "Assistente de Logística",
    "8": "Analista de Logística",
    "9": "Coordenador(a) de Logística",
    "10": "Assistente de RH",
    "11": "Analista de RH",
    "12": "Coordenador(a) de RH",
    "13": "Técnico(a) de Segurança do Trabalho",
    "14": "Engenheiro(a) de Segurança",
    "15": "Coordenador(a) de Segurança",
    "16": "Assistente Administrativo(a)",
    "17": "Analista Administrativo(a)",
    "18": "Coordenador(a) Administrativo(a)",
    "19": "Gerente Geral"
}

# limpar as colunas
df['area'] = (df['area'].astype(str).str.replace('.0', '', regex=False).str.strip())
df['cargo'] = (df['cargo'].astype(str).str.replace('.0', '', regex=False).str.strip())

# aplicar o mapeamento
df['area'] = df['area'].replace(map_area)
df['cargo'] = df['cargo'].replace(map_cargo)

# remover a coluna original
df = df.drop(columns=['Insira abaixo o código referente a seu cargo e área'])

# ver o resultado. 
df[['area', 'cargo']].head()

,area,cargo
0,Operação,Operador(a) de Produção
1,Manutenção,Supervisor(a) de Manutenção
2,Operação,Operador(a) de Produção
3,Operação,Operador(a) de Produção
4,Manutenção,Supervisor(a) de Manutenção


#### 2: Limpeza dos Dados

In [24]:
# verificar se há duplicatas na coluna 'ID de resposta'
df['ID de resposta'].duplicated().sum()

np.int64(0)

In [25]:
# Contar valores nulos por coluna e ordenar do maior para o menor
null_counts = df.isnull().sum().sort_values(ascending=False)

# Exibir apenas as colunas que têm pelo menos um valor nulo
print(null_counts[null_counts > 0])

Outro                                                                                 473
Dores nas pernas                                                                       55
Insensibilidade em relação aos colegas                                                 55
Vontade de ficar sozinho                                                               55
Impaciência com as pessoas em geral                                                    55
                                                                                     ... 
As tarefas que executo em meu trabalho são variadas                                    36
A avaliação do meu trabalho inclui aspectos além da minha produção                     36
Tenho autonomia para realizar as tarefas como julgo melhor                             36
As orientações que me são passadas para realizar as tarefas são coerentes entre si     36
Há clareza na definição das tarefas                                                    36
Length: 92

In [26]:
# remover coluna 'Outro' pois todos os dados dessa coluna são nulos
colunas_irrelevantes = ['Outro']
df = df.drop(columns=colunas_irrelevantes)

In [27]:
#verificar os dados da coluna 'Status de resposta' para entender quais são os status presentes
print("\n STATUS DE RESPOSTA:")
print(df['Status de resposta'].value_counts())
print(f"\nTotal: {len(df)} registros")


 STATUS DE RESPOSTA:
Status de resposta
Completed    418
Started       55
Name: count, dtype: int64

Total: 473 registros


418 usuarios responderam a pesquisa de forma completa e 55 ainda não completou. Com isso, vou fazer um filtro para analise de apenas os dados completos.

In [28]:
# Filtrar apenas respostas completas
df_completed = df.query('`Status de resposta` == "Completed"')
df_completed.head()

,ID de resposta,Status de resposta,Turno,Gênero,Cor/raça autodeclarada,Faixa etária,Escolaridade,Tempo na empresa,Tempo na função atual dentro da empresa,O número de trabalhadores é suficiente para a execução das tarefas,...,Dores nas costas,Dores no corpo,Dores no braço,Distúrbios Circulatórios,Dor de cabeça,Distúrbios digestivos,Alterações no sono,Alterações no apetite,area,cargo
1,210879164,Completed,Administrativo,Homem,Parda,35 a 44 anos,Pós graduação completa,De 1 a 3 anos,De 1 a 3 anos,4.0,...,2.0,2.0,2.0,1.0,2.0,2.0,1.0,1.0,Manutenção,Supervisor(a) de Manutenção
4,210888826,Completed,Administrativo,Mulher,Parda,35 a 44 anos,Pós graduação completa,De 1 a 3 anos,De 1 a 3 anos,3.0,...,3.0,2.0,2.0,3.0,2.0,2.0,3.0,3.0,Manutenção,Supervisor(a) de Manutenção
5,210889367,Completed,Administrativo,Mulher,Parda,19 a 24 anos,Ensino superior completo,De 1 a 3 anos,Menos de 1 ano,4.0,...,1.0,1.0,1.0,1.0,4.0,3.0,3.0,3.0,Manutenção,Supervisor(a) de Manutenção
7,210890156,Completed,Administrativo,Homem,Parda,25 a 34 anos,Ensino superior completo,De 1 a 3 anos,De 1 a 3 anos,4.0,...,4.0,2.0,2.0,2.0,3.0,4.0,3.0,3.0,Manutenção,Supervisor(a) de Manutenção
9,210894617,Completed,Administrativo,Mulher,Branca,19 a 24 anos,Ensino superior completo,De 1 a 3 anos,De 1 a 3 anos,4.0,...,1.0,1.0,1.0,1.0,3.0,1.0,1.0,1.0,Manutenção,Supervisor(a) de Manutenção


In [29]:
# Verificar se ainda existem valores nulos na base filtrada
print(df_completed.isnull().values.any())

False


In [30]:
# Reorganizar colunas para colocar as categóricas no início e as numéricas no final

#  Seleciona colunas categóricas
cat_cols = df_completed.select_dtypes(include=['object', 'category']).columns

# Seleciona demais colunas
num_cols = df_completed.select_dtypes(exclude=['object', 'category']).columns

# Reorganiza o dataframe
df_completed = df_completed[list(cat_cols) + list(num_cols)]

# Coloca 'ID de resposta' como primeira coluna
cols = ['ID de resposta'] + [
    col for col in df_completed.columns
    if col != 'ID de resposta'
]

df_completed = df_completed[cols]

In [31]:
df_completed.head()

,ID de resposta,Status de resposta,Turno,Gênero,Cor/raça autodeclarada,Faixa etária,Escolaridade,Tempo na empresa,Tempo na função atual dentro da empresa,area,...,Insensibilidade em relação aos colegas,Dores nas pernas,Dores nas costas,Dores no corpo,Dores no braço,Distúrbios Circulatórios,Dor de cabeça,Distúrbios digestivos,Alterações no sono,Alterações no apetite
1,210879164,Completed,Administrativo,Homem,Parda,35 a 44 anos,Pós graduação completa,De 1 a 3 anos,De 1 a 3 anos,Manutenção,...,1.0,2.0,2.0,2.0,2.0,1.0,2.0,2.0,1.0,1.0
4,210888826,Completed,Administrativo,Mulher,Parda,35 a 44 anos,Pós graduação completa,De 1 a 3 anos,De 1 a 3 anos,Manutenção,...,3.0,4.0,3.0,2.0,2.0,3.0,2.0,2.0,3.0,3.0
5,210889367,Completed,Administrativo,Mulher,Parda,19 a 24 anos,Ensino superior completo,De 1 a 3 anos,Menos de 1 ano,Manutenção,...,3.0,1.0,1.0,1.0,1.0,1.0,4.0,3.0,3.0,3.0
7,210890156,Completed,Administrativo,Homem,Parda,25 a 34 anos,Ensino superior completo,De 1 a 3 anos,De 1 a 3 anos,Manutenção,...,1.0,3.0,4.0,2.0,2.0,2.0,3.0,4.0,3.0,3.0
9,210894617,Completed,Administrativo,Mulher,Branca,19 a 24 anos,Ensino superior completo,De 1 a 3 anos,De 1 a 3 anos,Manutenção,...,1.0,1.0,1.0,1.0,1.0,1.0,3.0,1.0,1.0,1.0


## ETAPA para Mapeamento de Escalas PROART
##### Escala de resposta utilizada nas perguntas 9 a 99

1. Nunca
2. Raramente
3. Às vezes
4. Frequentemente
5. Sempre

In [32]:
# Definir questões por fator (Escalas PROART)
questoes = {
    'Organização Prescrita - Divisão de Tarefas': [
        'O número de trabalhadores é suficiente para a execução das tarefas',
        'Os recursos de trabalho são em número suficiente para a realização das tarefas',
        'Os equipamentos são adequados para a realização das tarefas',
        'O espaço físico disponível para a realização do trabalho é adequado',
        'O ritmo de trabalho é adequado',
        'Possuo condições adequadas para alcançar os resultados esperados do meu trabalho',
        'Os prazos para a realização das tarefas são flexíveis'
    ],
    'Organização Prescrita - Divisão Social': [
        'A comunicação entre chefe e subordinado é adequada',
        'As orientações que me são passadas para realizar as tarefas são coerentes entre si',
        'Tenho liberdade para opinar sobre o meu trabalho',
        'Os funcionários participam das decisões sobre o trabalho',
        'Tenho autonomia para realizar as tarefas como julgo melhor',
        'As informações de que preciso para executar minhas tarefas são claras',
        'Há qualidade na comunicação entre os funcionários',
        'Há justiça na distribuição das tarefas',
        'Há clareza na definição das tarefas',
        'Há flexibilidade nas normas para a execução das tarefas',
        'A avaliação do meu trabalho inclui aspectos além da minha produção',
        'As tarefas que executo em meu trabalho são variadas'
    ],
    'Estilos de Gestão - Individualista': [
        'Nesta organização os gestores se consideram o centro do mundo',
        'Os gestores desta organização se consideram insubstituíveis',
        'Os gestores desta organização fazem qualquer coisa para chamar a atenção',
        'Aqui os gestores preferem trabalhar individualmente',
        'Em meu trabalho, incentiva-se a idolatria dos chefes',
        'Há forte controle do trabalho',
        'A hierarquia é valorizada nesta organização',
        'O ambiente de trabalho se desorganiza com mudanças',
        'Os laços afetivos são fracos entre as pessoas desta organização',
        'É creditada grande importância para as regras nesta organização'
    ],
    'Estilos de Gestão - Coletivista': [
        'O trabalho coletivo é valorizado pelos gestores',
        'A competência dos trabalhadores é valorizada pela gestão',
        'Os gestores se preocupam com o bem estar dos trabalhadores',
        'Para esta organização, o resultado do trabalho é visto como uma realização do grupo',
        'O mérito das conquistas na empresa é de todos',
        'Somos incentivados pelos gestores a buscar novos desafios',
        'A inovação é valorizada nesta organização',
        'As decisões nesta organização são tomadas em grupo',
        'Os gestores favorecem o trabalho interativo de profissionais de diferentes áreas.',
        'Existem oportunidades semelhante de ascensão para todas as pessoas',
        'As pessoas são compromissadas com a organização mesmo quando não há retorno adequado'
    ],
    'Sofrimento - Falta de Sentido': [
        'Meu trabalho é sem sentido',
        'Meu trabalho é irrelevante para o desenvolvimento da sociedade',
        'Minhas tarefas são banais',
        'Considero minhas tarefas insignificantes',
        'Sinto-me improdutivo no meu trabalho',
        'A identificação com minhas tarefas é inexistente',
        'Sinto-me inútil em meu trabalho',
        'Sinto-me desmotivado para realizar minhas tarefas',
        'Permaneço neste emprego por falta de oportunidade no mercado trabalho'
    ],
    'Sofrimento - Esgotamento Mental': [
        'Meu trabalho é desgastante',
        'Meu trabalho é cansativo',
        'Meu trabalho me sobrecarrega',
        'Meu trabalho me desanima',
        'Meu trabalho me frustra',
        'Meu trabalho me faz sofrer',
        'Meu trabalho me causa insatisfação',
        'Submeter meu trabalho a decisões políticas é fonte de revolta'
    ],
    'Sofrimento - Falta de Reconhecimento': [
        'Falta-me liberdade para dialogar com minha chefia',
        'Minha chefia trata meu trabalho com indiferença',
        'Há desconfiança na relação entre chefia e subordinado',
        'O trabalho que realizo é desqualificado pela chefia',
        'Sou excluído do planejamento de minhas próprias tarefas',
        'Falta-me liberdade para dizer o que penso sobre meu trabalho',
        'Meus colegas são indiferentes comigo',
        'Meus colegas desvalorizam meu trabalho',
        'É difícil a convivência com meus colegas',
        'A submissão do meu chefe às ordens superiores me causa revolta',
        'Meu trabalho é desvalorizado pela organização'
    ],
    'Danos - Psicológicos': [
        'Sensação de vazio',
        'Amargura',
        'Tristeza',
        'Vontade de Desistir de Tudo',
        'Perda da autoconfiança',
        'Solidão'
    ],
    'Danos - Sociais': [
        'Mau-Humor',
        'Dificuldade com os amigos',
        'Conflitos nas relações familiares',
        'Agressividade com os outros',
        'Dificuldades nas relações fora do trabalho',
        'Impaciência com as pessoas em geral',
        'Vontade de ficar sozinho',
        'Insensibilidade em relação aos colegas'
    ],
    'Danos - Físicos': [
        'Dores nas pernas',
        'Dores nas costas',
        'Dores no corpo',
        'Dores no braço',
        'Distúrbios Circulatórios',
        'Dor de cabeça',
        'Distúrbios digestivos',
        'Alterações no sono',
        'Alterações no apetite'
    ]
}

print(f" MAPEAMENTO DE ESCALAS:")
print(f"   Total de fatores: {len(questoes)}")
for fator, qs in questoes.items():
    print(f"   • {fator}: {len(qs)} questões")

 MAPEAMENTO DE ESCALAS:
   Total de fatores: 10
   • Organização Prescrita - Divisão de Tarefas: 7 questões
   • Organização Prescrita - Divisão Social: 12 questões
   • Estilos de Gestão - Individualista: 10 questões
   • Estilos de Gestão - Coletivista: 11 questões
   • Sofrimento - Falta de Sentido: 9 questões
   • Sofrimento - Esgotamento Mental: 8 questões
   • Sofrimento - Falta de Reconhecimento: 11 questões
   • Danos - Psicológicos: 6 questões
   • Danos - Sociais: 8 questões
   • Danos - Físicos: 9 questões


##### Salvar aba com os fatores
Salvamento do mapeamento em outra aba do excel dos dados tratados para o PowerBI

In [36]:
df_completed.to_excel('output/dados-tratados.xlsx', index=False)

# Criar DataFrame de mapeamento Fator-Questão para Power BI
fatores = []
questoes_list = []
for fator, lista in questoes.items():
    for q in lista:
        fatores.append(fator)
        questoes_list.append(q)
df_fatores_questoes = pd.DataFrame({'Fator': fatores, 'Questao': questoes_list})

with pd.ExcelWriter('output/dados-tratados.xlsx', mode='a', engine='openpyxl', if_sheet_exists='replace') as writer:
    df_fatores_questoes.to_excel(writer, sheet_name='Fatores_Questoes', index=False)
print("Sheet 'Fatores_Questoes' adicionada ao arquivo dados-tratados.xlsx!")

Sheet 'Fatores_Questoes' adicionada ao arquivo dados-tratados.xlsx!


Dados prontos para analises no PowerBI 